enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,348 @@
+multiple values.
+
+### Machine Config Operator
+
+No changes are needed in the MCO for Phase 1 as the machine-os-content image would be a manifest listed image and the machine config daemon would
+extract the relevant architecture's machine-os-content based on the node it runs on. In the future, architecture specific configurations should
+not be required, instead they need to be templatized and be generalized for certain items, for example kubelet system reserved memory could scale
+based on page size.
+
+### Hypershift
+
+The addition of heterogeneous support in Openshift also opens up a new dimension for [support][OCPPLAN-5684] in Hypershift where the control plane 
+and the workers could run on different architectures. There would be certain changes needed to realize this as there are limitations which need to
+be addressed:
+
+- When creating the hosted cluster, we need to override the `--release-image` option to specify the manifest listed release image in order
+to be able to deploy on a worker node of any architecture. Optionally, the config could provide a flag to indicate that it is being installed
+on a heterogeneous architecture cluster in which case it will pull down a manifest listed release payload rather than the x86 release payload.
+- For the hosted control plane to work on non x86 architecture nodes, the following image references would need changing:
+  - the release image fixtures referencing x86 imagestreams. these would need to be changed to manifestlist references or there would need
+  to be imagestreams added for individual architectures and they would be picked based on the node architecture.
+  - references to some images like cluster-api-provider-aws, apiserver-network-proxy etc.. are x86. these would need to change to provide the images
+  for other architectures too. These images can be overridden through annotations today.
+- Another option would be to mandate an x86 worker node be present for Hypershift and nodeselect the x86 node for the hosted cluster control plane
+deployment.
+- When creating the nodepool, the defaults for instances types and AMIs are x86 specific. The instance types can be overridden but not the AMIs.
+An option needs to be added to the create cluster command to override AMIs.
+
+### Console
+
+As of today console filters content based on control plane architecture. This needs to be entirely removed or made configurable for
+heterogeneous clusters. The operator should be able to filter out the set of OLM operators that, if installed will be able to find
+worker nodes to run on. This means that it has to display only operators that can be supported by architectures that the workers are
+running on. Additionally, it would also have to tell OLM which architectures to nodeselect on by creating a subscription object and
+adding a podAffinity rule with nodeselectorterms. The support for podAffinity in the subscription object will be added to OLM.
```

Comment:
user should be able to choose which architectures are put into the nodeaffinity on the subscription.  but the console should default it to whatever the operator supports (user may wish to deselect some architectures).

also, a nit, i'd make this a separate paragraph because they are two different features the console needs to build and people might gloss over it as it is currently written.

Reply:
updated.